File: src/test/java/info/dreamcoder/DBStringTest.java
package info.dreamcoder;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

@DisplayName("测试字符串类型的列")
class DBStringTest {

    @Test
    @DisplayName("能正确的创建字符串的类型，默认是255")
    void shouldCreateStringColumn() {
        DBString col = new DBString("string_column");
        assertEquals(col.toSql(), "string_column varchar(255)");
    }
}

File: migration-core/src/test/kotlin/columns/DbStringTest.kt
package columns

import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class DbStringTest {

    private lateinit var column: DbString

    @BeforeEach
    fun initColumn() {
        column = DbString("test_column");
    }

    @Test
    @DisplayName("默认生成250 varchar类型的列")
    fun shouldCreateVarcharColumnWith250Length() {
        assertEquals(
            column.toSql(),
            "test_column varchar(250)"
        )
    }

    @Test
    @DisplayName("能修改varchar类型的长度")
    fun canChangeColumnLength() {
        column.limit(100)
        assertEquals(
            column.toSql(),
            "test_column varchar(100)"
        )
    }
}
